src/Microsoft.DotNet.Interactive/Connection/KernelCommandEnvelope.cs

```diff
@@ -187,7 +187,8 @@ public static IKernelCommandEnvelope Deserialize(JsonElement json)
             {
                 var uri = new Uri(routingSlipItem.GetString(), UriKind.Absolute);
 
-                command.TryAddToRoutingSlip(uri);
+                command.RoutingSlip.StampAsArrived(uri);
+                command.RoutingSlip.Stamp(uri);
```

Comment:
This looks wrong. I would think that on deserialization, a command will frequently not be completed yet. 

Can these two states be distinguished after deserialization?